core/src/cost_update_service.rs

```diff
@@ -132,17 +132,16 @@ impl CostUpdateService {
                     CostUpdate::ExecuteTiming {
                         mut execute_timings,
                     } => {
-                        dirty |= Self::update_cost_model(&cost_model, &mut execute_timings);
+                        updated_program_costs =
+                            Self::update_cost_model(&cost_model, &mut execute_timings);
```

Comment:
shouldn't this actually extend updated_program_costs rather than assign to it?
looks like each iteration of cost_update loop is overwriting updated_program_costs before previous ones are sent to persist_cost_table

Reply:
So glad you pointed it out. This code path needs to be reworked - the logic is wrong, unnecessarily allocating HashMaps, etc. Working on an update